[Loh Jin Hun] IP

Ragnapop opened the pull request.

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

Ragnapop committed:
Level 0. Rename, Greet, Exit

Ragnapop committed:
Level 1. Echo

Ragnapop committed:
Level 3. Mark as Done

Ragnapop committed:
Level 4. ToDos, Events, Deadlines

Ragnapop committed:
Level 5. Handle Errors

Ragnapop committed:
Level 6. Delete

Ragnapop committed:
Update Duke.java

Ragnapop committed:
update

Ragnapop committed:
UI Testing

Ragnapop committed:
Level 7. Save

Ragnapop committed:
Merge pull request #1 from Ragnapop/branch-Level-7

Level 7. Save

Ragnapop committed:
Level 8. Dates and Times

Ragnapop committed:
OOP changes

Ragnapop committed:
Merge pull request #2 from Ragnapop/branch-OOP

OOP changes

Ragnapop committed:
Merge branch 'master' into branch-level-8

Ragnapop committed:
Merge pull request #3 from Ragnapop/branch-level-8

Level 8. Dates and Times & OOP changes

Ragnapop committed:
packaging

Ragnapop committed:
Merge pull request #4 from Ragnapop/branch-package

packaging

Ragnapop committed:
Merge branch 'add-gradle-support'

Ragnapop committed:
minor commits for Jtest

Ragnapop committed:
Update EventTest.java

Ragnapop committed:
A-JUnit

Ragnapop committed:
A-JavaDoc

Ragnapop committed:
Merge pull request #5 from Ragnapop/branch-A-JavaDoc

A-JavaDoc

Ragnapop committed:
Level 9. Find

Ragnapop committed:
Merge pull request #6 from Ragnapop/branch-level-9

Level 9. Find

Ragnapop committed:
A-CodingStandard

Ragnapop committed:
Merge pull request #7 from Ragnapop/branch-A-CodingStandard

A-CodingStandard

guanquann commented on build.gradle:
```diff
@@ -0,0 +1,43 @@
+}
+
+application {
+    mainClass.set("seedu.duke.Duke")
```
```suggestion
    mainClass.set("duke.Duke")
```

You need to follow your package name in order for gradle `gradlew run` command to work

guanquann commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,53 @@
+    public static void intro() {
+        System.out.println("Hello from\n" + BINGUS_LOGO);
+        System.out.println("Hello! I'm \n" + NAME);
+        System.out.println("What can I do for you?\n");// initial introductory message
```
It looks like you are repeating your comment. The javadocs for this method already states what the message does.

Same issue with methods such as `bye()`

guanquann commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,203 @@
+     * @param str The string representation of the event.
+     */
+    public static void eventCase(String str) throws DukeException {
+        str = str.replace("event", "");
```
According to the documentation, you should not reuse formal parameters as local variables inside the method.

```suggestion
        String parsedString = str.replace("event", "");
```

guanquann commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,203 @@
+        String keyword = str.replace("find", "");
+        System.out.println("Here are the matching tasks in your list:");
+        for (int i = 0; i < list.size(); i++) {
+            int n = 0;
```
Your variable name should be more detailed. I assume this counts the number of tasks containing the keywords? 

```suggestion
            int caseCount = 0;
```

guanquann reviewed (commented):
Most of the code looks good to me! Some minor issues with your code quality. But do take a look at your coding standards (I will leave this part to your first reviewer) :D

Ragnapop committed:
A-CheckStyle

Ragnapop committed:
Update duke.txt

Ragnapop committed:
Merge pull request #8 from Ragnapop/A-CheckStyle

A-CheckStyle

Ragnapop committed:
Level-10

added GUI

Ragnapop committed:
Merge branch 'master' into branch-Level-10

Ragnapop committed:
Merge pull request #9 from Ragnapop/branch-Level-10

Level 10. GUI

Ragnapop committed:
Update README.md

Ragnapop committed:
Added exit messages

Ragnapop committed:
Merge branch 'branch-Level-10' of https://github.com/Ragnapop/ip into branch-Level-10

Ragnapop committed:
Merge pull request #10 from Ragnapop/branch-Level-10

Branch level 10

chengjunyuan commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,63 @@
+        * returns string representation of task.
+        *
+        */
+        public String ToString(){
```
Would be good to change to `toString` instead of `ToString`. Take note that doing so would also mean overriding the default `toString` object method.

chengjunyuan commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,63 @@
+        *
+        */
+        public void unmarkAsDone(){
+        isDone = false;
```
The preview on Github shows some strange indentation here? Seems to me that the last brace `}` is pushed too far to the left, and so is the line `isDone = false;`

chengjunyuan commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,63 @@
+        *
+        */
+        public String toStore(){
+            if(isDone) {
```
Add spacing, i.e., `if (isDone)` instead of `if(isDone)`.

chengjunyuan commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,242 @@
+
+       //System.out.println("Now you have " + list.size() + " tasks in the list.");
+        store.save();
+        return "Got it. I've added this task:\n" + list.get(list.size()-1).ToString() + "\n" + "Now you have " + list.size() + " tasks in the list.\n";
```
Indentation looks off in the Github preview - would be good to check on your end and make sure it's proper.

chengjunyuan commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,242 @@
+            return "OOPS!!! The description of a event cannot be empty." + "Please give this instruction in the following format: event [description] / [event starting date] / [event ending date]\n";
+            //throw new DukeException("OOPS!!! The beginning date of a event cannot be empty." +
+                    //"Please give this instruction in the following format: event [description] / [event starting date] / [event ending date]");
+        } else if(eventtokens.length < 3) {
```
Add spacing between `if` and condition within bracket

chengjunyuan commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,42 @@
+               parser.parse(str);
+
+
+
```
Are these empty lines necessary? Would be good to double check, sometimes all the IntelliJ highlighting and linting makes line spacing render weirdly, so this might be caused by that.

chengjunyuan commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,81 @@
+ *  * Represents the main class of the application.
+ *
+ *
+ * @author Loh Jin Hun
```
Good use of JavaDoc, well done!

chengjunyuan commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,81 @@
+    private static Parser parser;
+    private static final String FILE_PATH = "data/duke.txt";
+    final static String BINGUS_LOGO = " B   i   n   g   u   s ";
+    final static  String NAME = "Bingus";
```
Extra spacing

Ragnapop replied:
Thanks!

Ragnapop committed:
Update README.md

Ragnapop committed:
Update build.gradle

Co-authored-by: Guan Quan <76832850+guanquann@users.noreply.github.com>

Ragnapop committed:
Update src/main/java/duke/TaskList.java

Co-authored-by: Guan Quan <76832850+guanquann@users.noreply.github.com>

Ragnapop committed:
Update README.md

Ragnapop committed:
Add assertions

Program has no assertions and thus cannot ensure the validity of states.

This may cause the program to run with unintended errors.

Assertions are added, mainly in TaskList.java and Main.java.

This is to ensure early termination due to errors in the process.

Ragnapop committed:
A-CodeQuality

Ragnapop committed:
Merge pull request #11 from Ragnapop/branch-A-Assertions

Add assertions

Ragnapop committed:
Merge branch 'master' into branch-A-CodeQuality

Ragnapop committed:
Merge pull request #12 from Ragnapop/branch-A-CodeQuality

A-CodeQuality

Ragnapop committed:
C-Priority

There is currently no method to separate tasks based on priority.

Added different priority levels for users to differentiate their tasks.

Levels range from low, medium to high. A default task has a priority level of None is given to tasks that do not have a priority level allocated to them

Ragnapop committed:
Merge pull request #13 from Ragnapop/branch-BCD-Extension

C-Priority

Ragnapop committed:
Added Ui.png

Ragnapop committed:
minor GUI changes

Ragnapop committed:
Merge branch 'master' of https://github.com/Ragnapop/ip

Ragnapop committed:
minor test case changes

Ragnapop committed:
added Ui.png

Ragnapop committed:
Update README.md

Ragnapop committed:
small adjustments

Ragnapop committed:
Merge branch 'master' of https://github.com/Ragnapop/ip

Ragnapop committed:
minor changes

Ragnapop committed:
minor changes

Ragnapop committed:
Update TaskList.java
